### ISSUE TITLE
Evaluation Method is chosen improperly chosen

### ISSUE THREAD

**Issue by NicoHambauer** [Zaharah/processtransformer#1]
Despite there is a mistake in Line 102 where accuracy is only a single value and no the list which is named accuracies, this evaluation method might be improperly chosen, since it calculates the mean of the accuracies for each prefix size and averages it afterward but does not take into account support numbers of these samples. 

[See referred code here](https://github.com/Zaharah/processtransformer/blob/178f4c0bde5efb6bb25834d3493841482ad58227/next_activity.py#L85-L105)

This means the test set is divided by the length of each trace/its prefix size ( = none zero values in sample x) and then regardless of the amount of these samples it is averaged (unweighted average). If you would take a simple and known approach, by just evaluating the whole test dataset at once, and compare it to this method, the simple and correct approach metrics would be lower. This method might artificially push metrics in your paper and implementation, like accuracy, fscore, precision, and recall. I tested it and it was differing by 5-8 percentage points depending on the eventlog in accuracy for example. This is quite a lot in a PBPM evaluation, regarding a comparison like you did with baselines, like Tax et al.

Maybe I am wrong and of course this is one way of doing it, but especially for helpdesl dataset this is leading to a very distorted result, since helpdesk includes a few very rare samples in the log, would like to check on this issue and learn from it if I am mistaken!

I did a quick evaluation with the setting you provide in this repository and then did a reference evaluation by leaving out the split of the test set by prefix size and instead tested on the whole test set. Results below:

Results with the current setting of the repo:
![image](https://user-images.githubusercontent.com/58194361/123064047-39d7b280-d40e-11eb-85e4-0b337b395840.png)

Results adjusted to match the evaluation method of the baselines mentioned in the paper:
![image](https://user-images.githubusercontent.com/58194361/123066115-0eee5e00-d410-11eb-9824-b0ce8bb0b585.png)

So as can be derived from the images, changing to the weighted average calculation, which the other baselines mentioned in the paper actually should do, changes results in metrics.

I also plotted/saved some details on the test dataset, which was split by prefix size in your code and each metric is then averaged (unweighted) in your code again:
![image](https://user-images.githubusercontent.com/58194361/123066667-9340e100-d410-11eb-8f7e-8b1fb8f87b24.png)

You are also mentioning in the [paper](https://arxiv.org/abs/2104.00721) that
> Additionally, 20% of the data from the training
> split is used for validation and hyper-parameter tuning during the learning phase.

which I do not see in the current git repository (speaking of next activity prediction).
Maybe the latest release is just not pushed on github?

**Comment by Zaharah** [comment]
The accuracy is typo. You are correct there. 
For the evaluation method, it is a deliberate design choice which is also clearly mentioned in the paper.  This is not a  mistake. You can evaluate the model with the method of your choice.
